From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: ZHANG Yuntian <95260730+RadxaYuntian@users.noreply.github.com>
Date: Thu, 25 Aug 2022 19:53:56 +0800
Subject: [PATCH] Include cma_heap_get_phys when needed

---
 drivers/dma-buf/heaps/cma_heap.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/dma-buf/heaps/cma_heap.c b/drivers/dma-buf/heaps/cma_heap.c
index ab6f937eb6..803f523cd3 100644
--- a/drivers/dma-buf/heaps/cma_heap.c
+++ b/drivers/dma-buf/heaps/cma_heap.c
@@ -441,6 +441,7 @@ static struct dma_buf *cma_heap_allocate(struct dma_heap *heap,
 	return cma_heap_do_allocate(heap, len, fd_flags, heap_flags, false);
 }
 
+#if IS_ENABLED(CONFIG_NO_GKI)
 static int cma_heap_get_phys(struct dma_heap *heap,
 			     struct dma_heap_phys_data *phys)
 {
@@ -471,6 +472,7 @@ static int cma_heap_get_phys(struct dma_heap *heap,
 
 	return (phys->paddr == (__u64)-1) ? -EINVAL : 0;
 }
+#endif
 
 static const struct dma_heap_ops cma_heap_ops = {
 	.allocate = cma_heap_allocate,
-- 
2.37.2

