From 51361929254412cf900c6d029b5b567248b39c44 Mon Sep 17 00:00:00 2001
From: bsp <bsp@radxa.com>
Date: Wed, 29 Nov 2023 19:57:03 +0800
Subject: [PATCH 1/2] enable i2s_8ch_0 and modify uart0 pinctrl

Signed-off-by: Ken Wang <ken@radxa.com>

diff --git a/arch/arm64/boot/dts/rockchip/rk3308-radxa-s0.dts b/arch/arm64/boot/dts/rockchip/rk3308-radxa-s0.dts
index ee1a1efdef..473e636445 100644
--- a/arch/arm64/boot/dts/rockchip/rk3308-radxa-s0.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3308-radxa-s0.dts
@@ -44,7 +44,7 @@ codec: acodec-sound {
 			simple-audio-card,widgets =
 				"Headphone", "Headphones";
 			simple-audio-card,cpu {
-				sound-dai = <&i2s_8ch_2>;
+				sound-dai = <&i2s_8ch_0>;
 			};
 			simple-audio-card,codec {
 				sound-dai = <&acodec>;
@@ -290,10 +290,11 @@ &i2s_8ch_0 {
 	assigned-clock-parents = <&cru SCLK_I2S0_8CH_TX_MUX>;
 	rockchip,clk-trcm = <1>;
 	#sound-dai-cells = <0>;
+	status = "okay";
 };
 
 &i2s_8ch_2 {
-	status = "okay";
+	status = "disabled";
 	#sound-dai-cells = <0>;
 };
 
@@ -373,6 +374,8 @@ &spi2 {
 
 &uart0 {
 	status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 = <&uart0_xfer>;
 };
 
 &uart2 {
-- 
2.25.1

