From aacf20cf30d59cfe5df9716d8612a65498198d92 Mon Sep 17 00:00:00 2001
From: bsp <bsp@radxa.com>
Date: Fri, 23 Dec 2022 05:59:26 +0800
Subject: [PATCH 2/2] arm: dts: rock 4: update vdd_cpu_b and vdd_gpu

Signed-off-by: bsp <bsp@radxa.com>

diff --git a/arch/arm/dts/rk3399-rock-pi-4.dtsi b/arch/arm/dts/rk3399-rock-pi-4.dtsi
index dbf49fd109..0ba474cbd9 100644
--- a/arch/arm/dts/rk3399-rock-pi-4.dtsi
+++ b/arch/arm/dts/rk3399-rock-pi-4.dtsi
@@ -421,38 +421,43 @@
 	};
 
 	vdd_cpu_b: regulator@40 {
+		u-boot,dm-spl;
 		compatible = "silergy,syr827";
 		reg = <0x40>;
 		fcs,suspend-voltage-selector = <1>;
-		pinctrl-names = "default";
+		regulator-compatible = "fan53555-reg";
 		pinctrl-0 = <&vsel1_pin>;
+		vsel-gpios = <&gpio1 RK_PC1 GPIO_ACTIVE_HIGH>;
 		regulator-name = "vdd_cpu_b";
 		regulator-min-microvolt = <712500>;
 		regulator-max-microvolt = <1500000>;
+		regulator-init-microvolt = <900000>;
 		regulator-ramp-delay = <1000>;
 		regulator-always-on;
 		regulator-boot-on;
 		vin-supply = <&vcc5v0_sys>;
-
 		regulator-state-mem {
 			regulator-off-in-suspend;
 		};
 	};
 
 	vdd_gpu: regulator@41 {
+		u-boot,dm-spl;
 		compatible = "silergy,syr828";
 		reg = <0x41>;
 		fcs,suspend-voltage-selector = <1>;
-		pinctrl-names = "default";
+		regulator-compatible = "fan53555-reg";
 		pinctrl-0 = <&vsel2_pin>;
+		vsel-gpios = <&gpio1 RK_PB6 GPIO_ACTIVE_HIGH>;
 		regulator-name = "vdd_gpu";
 		regulator-min-microvolt = <712500>;
 		regulator-max-microvolt = <1500000>;
+		regulator-init-microvolt = <900000>;
 		regulator-ramp-delay = <1000>;
 		regulator-always-on;
 		regulator-boot-on;
 		vin-supply = <&vcc5v0_sys>;
-
+		regulator-initial-mode = <1>; /* 1:force PWM 2:auto */
 		regulator-state-mem {
 			regulator-off-in-suspend;
 		};
-- 
2.30.2

