From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Yuntian Zhang <yt@radxa.com>
Date: Tue, 26 Apr 2022 14:24:30 +0800
Subject: [PATCH] rockchip: add ROCK 5A support

Signed-off-by: Yuntian Zhang <yt@radxa.com>
---
 arch/arm/dts/Makefile             |   1 +
 1 files changed, 1 insertions(+)

diff --git a/arch/arm/dts/Makefile b/arch/arm/dts/Makefile
index b834bc3fc3..05a1a5967b 100644
--- a/arch/arm/dts/Makefile
+++ b/arch/arm/dts/Makefile
@@ -56,6 +56,7 @@ dtb-$(CONFIG_ARCH_ROCKCHIP) += \
 	rk3399-puma-ddr1333.dtb \
 	rk3399-puma-ddr1600.dtb \
 	rk3399-puma-ddr1866.dtb \
+	rk3588s-rock-5a.dtb \
 	rv1108-evb.dtb
 dtb-$(CONFIG_ARCH_MESON) += \
 	meson-gxbb-odroidc2.dtb
-- 
2.36.1

