From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: bsp <bsp@radxa.com>
Date: Mon, 29 May 2023 18:11:30 +0800
Subject: [PATCH] HACK: retry for USB storage devices

---
 common/usb_storage.c | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/common/usb_storage.c b/common/usb_storage.c
index ac642757737..d046d647c6b 100644
--- a/common/usb_storage.c
+++ b/common/usb_storage.c
@@ -1481,11 +1481,18 @@ int usb_stor_get_info(struct usb_device *dev, struct us_data *ss,
 #endif /* CONFIG_USB_BIN_FIXUP */
 	debug("ISO Vers %X, Response Data %X\n", usb_stor_buf[2],
 	      usb_stor_buf[3]);
-	if (usb_test_unit_ready(pccb, ss)) {
+	for (int i = 0; usb_test_unit_ready(pccb, ss); i++) {
 		printf("Device NOT ready\n"
 		       "   Request Sense returned %02X %02X %02X\n",
 		       pccb->sense_buf[2], pccb->sense_buf[12],
 		       pccb->sense_buf[13]);
+		if (pccb->sense_buf[2] == 0x02 && // Not Ready Sense Key
+		    pccb->sense_buf[12] == 0x04 && // Not Ready, In Process of Becoming Ready
+			pccb->sense_buf[13] == 0x01 &&
+			i < 5) {
+			udelay(200 * 1000);
+			continue;
+		}
 		if (dev_desc->removable == 1)
 			dev_desc->type = perq;
 		return 0;
-- 
2.40.1

