From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: bsp <bsp@radxa.com>
Date: Tue, 18 Apr 2023 17:23:36 +0800
Subject: [PATCH] Add Radxa ZERO gpiod definitions

---
 .../dts/amlogic/meson-g12a-radxa-zero.dts     | 31 +++++++++++++++++++
 1 file changed, 31 insertions(+)

diff --git a/arch/arm64/boot/dts/amlogic/meson-g12a-radxa-zero.dts b/arch/arm64/boot/dts/amlogic/meson-g12a-radxa-zero.dts
index 51136681b1ca..d05f16f12a35 100644
--- a/arch/arm64/boot/dts/amlogic/meson-g12a-radxa-zero.dts
+++ b/arch/arm64/boot/dts/amlogic/meson-g12a-radxa-zero.dts
@@ -398,6 +398,37 @@ brcmf: wifi@1 {
 	};
 };
 
+&gpio {
+	gpio-line-names =
+		/* GPIOZ */
+		"", "", "", "", "", "", "", "",
+		"", "", "", "", "", "", "", "",
+		/* GPIOH */
+		"", "", "", "", "PIN_19", "PIN_21", "PIN_24", "PIN_23",
+		"PIN_36",
+		/* BOOT */
+		"", "", "", "", "", "", "", "",
+		"", "", "", "", "EMMC_PWRSEQ", "", "", "",
+		/* GPIOC */
+		"", "", "", "", "", "", "SD_CD", "PIN_22",
+		/* GPIOA */
+		"", "", "", "", "", "", "", "",
+		"", "", "", "", "", "", "PIN_3", "PIN_5",
+		/* GPIOX */
+		"", "", "", "", "", "", "SDIO_PWRSEQ", "",
+		"PIN_18", "PIN_12", "PIN_16", "PIN_13", "", "", "", "",
+		"", "BT_SHUTDOWN", "", "";
+};
+
+&gpio_ao {
+	gpio-line-names =
+		/* GPIOAO */
+		"PIN_8", "PIN_10", "PIN_11", "PIN_7", "PIN_32", "FUSB_IRQ", "", "",
+		"PIN_35", "PIN_37", "PIN_38", "PIN_11",
+		/* GPIOE */
+		"", "", "";
+};
+
 /* SD card */
 &sd_emmc_b {
 	status = "okay";
-- 
2.40.0

