From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Yuntian Zhang <yt@radxa.com>
Date: Thu, 28 Jul 2022 19:02:20 +0800
Subject: [PATCH] Initialized cnt before using

Signed-off-by: Yuntian Zhang <yt@radxa.com>
---
 drivers/power/rk818_battery.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/power/rk818_battery.c b/drivers/power/rk818_battery.c
index fbeb88a979..fd2e239f6c 100644
--- a/drivers/power/rk818_battery.c
+++ b/drivers/power/rk818_battery.c
@@ -1570,6 +1570,8 @@ static void rk818_bat_calc_zero_linek(struct rk818_battery *di)
 	int i, cnt, vol_old, vol_now;
 	int org_linek = 0, min_gap_xsoc;
 
+	cnt = 0;
+
 	if ((abs(di->current_avg) < 500) && (di->dsoc > 10))
 		pwroff_vol = di->pdata->pwroff_vol + 50;
 	else
-- 
2.37.1

