From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: ZHANG Yuntian <yt@radxa.com>
Date: Tue, 25 Jun 2024 16:31:45 +0800
Subject: [PATCH] fix: import namespace DMA_BUF before using it

---
 drivers/usb/gadget/function/uvc_queue.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/usb/gadget/function/uvc_queue.c b/drivers/usb/gadget/function/uvc_queue.c
index 3565bbe8976e1..237e2335cee6c 100644
--- a/drivers/usb/gadget/function/uvc_queue.c
+++ b/drivers/usb/gadget/function/uvc_queue.c
@@ -81,6 +81,7 @@ static int uvc_queue_setup(struct vb2_queue *vq,
 }
 
 #if defined(CONFIG_ARCH_ROCKCHIP) && defined(CONFIG_NO_GKI)
+MODULE_IMPORT_NS(DMA_BUF);
 /*
  * uvc_dma_buf_phys_to_virt - Get the physical address of the dma_buf and
  * translate it to virtual address.
-- 
2.45.2

